src/groups/circle_group_sphere.jl

```diff
@@ -0,0 +1,149 @@
+    )
+end
+
+function CircleGroup(::Euclidean{ManifoldsBase.TypeParameter{Tuple{2}},ℝ})
+    return CircleGroup(Sphere(1))
+end
+
+const _PlanarCircleGroup = LieGroup{
+    ℝ,AbelianMultiplicationGroupOperation,<:Sphere{ManifoldsBase.TypeParameter{Tuple{1}},ℝ}
+}
+
+#=
+function diff_conjugate!(::_PlanarCircleGroup, g, h, X)
+    return X
+end
+=#
+_doc_diff_left_compose_mult_planar_circ = """
+    diff_left_compose(G::LieGroup{ℝ, AbelianMultiplicationGroupOperation, Sphere}, g, h, X)
+    diff_left_compose!(G::LieGroup{ℝ, AbelianMultiplicationGroupOperation, Sphere}, Y, g, h, X)
+
+Compute the differential of the left group multiplication ``λ_g(h) = g$(_math(:∘))h`` of the Circle Group,
+represented as two dimensional vectors in ``ℝ^2``.
+
+It simplifies for the [`AbelianMultiplicationGroupOperation`](@ref) to ``Dλ_g(h)[X] = g*X``,
+where the multiplication corresponds to the complex multiplication after canonical identification
+of the real plane with the complex plane.
+
+This can be computed in-place of `Y`.
+"""
+
+@doc "$(_doc_diff_left_compose_mult_planar_circ)"
+diff_left_compose!(C::_PlanarCircleGroup, Y, g, h, X) = compose!(C, Y, g, X)
+
+@doc "$(_doc_diff_left_compose_mult_planar_circ)"
+diff_left_compose(_PlanarCircleGroup, g, h, X)
+
+_doc_diff_right_compose_mult_planar_circ = """
+    diff_right_compose(G::LieGroup{ℝ, AbelianMultiplicationGroupOperation, Sphere}, g, h, X)
+    diff_right_compose!(G::LieGroup{ℝ, AbelianMultiplicationGroupOperation, Sphere}, Y, g, h, X)
+
+Compute the differential of the right group multiplication ``λ_g(h) = g$(_math(:∘))h`` of the Circle Group,
+represented as two dimensional vectors in ``ℝ^2``.
+
+It simplifies for the [`AbelianMultiplicationGroupOperation`](@ref) to ``Dλ_g(h)[X] = X*g``,
+where the multiplication corresponds to the complex multiplication after canonical identification
+of the real plane with the complex plane.
+"""
+
+@doc "$(_doc_diff_right_compose_mult_planar_circ)"
+diff_right_compose!(C::_PlanarCircleGroup, Y, g, h, X) = compose!(C, Y, X, g)
+
+_doc_exp_planar_circ = """
+    exp(::LieGroup{ℝ, AbelianMultiplicationGroupOperation, Sphere}, X)
+    exp!(::LieGroup{ℝ, AbelianMultiplicationGroupOperation, Sphere}, g, X)
+
+Compute the Lie group exponential on the [`CircleGroup`](@ref), represented as two dimensional vectors in the real plane.
+It coincides with the [ordinary complex exponential](https://en.wikipedia.org/wiki/Exponential_map_(Lie_theory)#Examples) after
+canonical identification of the real plane with the complex plane.
+
+This can be computed in-place of `g`.
+```math
+$(_tex(:exp)) $(_tex(:pmatrix, 0, "t")) = $(_tex(:pmatrix, _tex(:cos)*"(t)", _tex(:sin)*"(t)"))
+```
+"""
+@doc "$(_doc_exp_planar_circ)"
+ManifoldsBase.exp(::_PlanarCircleGroup, X)
+
+@doc "$(_doc_exp_planar_circ)"
+function ManifoldsBase.exp!(::_PlanarCircleGroup, g, X)
+    z = exp(X[1] + X[2] * im)
+    g[1] = real(z)
+    g[2] = imag(z)
+    return nothing
```

Comment:
```suggestion
    return g
```
Mutating functions typically return the mutated arguments in JuliaManifolds.

Reply:
ouh! Well spotted!